File: python/graphs/Graph.py
class Graph:
    def __init__(self):
        self.vertices = []
        self.edges = []

    def add_edge(self, u, v):
        self.edges.append((u, v))

    def add_vertex(self, v):
        self.vertices.append(v)

    def remove_vertex(self, v):
        try:
            idx = self.vertices.index(v)
            del self.vertices[idx]
            return True
        except ValueError:
            return False

    def remove_edge(self, u, v):
        try:
            idx = self.edges.index((u, v))
            del self.edges[idx]
            return True
        except ValueError:
            return False

    def adjacent(self, u, v):
        return (u, v) in self.edges

File: java/main/java/eu/libal/cpptoolkit/graphs/Graph.java
package eu.libal.cpptoolkit.graphs;

import java.util.List;

public class Graph {
    public Graph() {
    }

    public <T> void addVertex(Vertex<T> v) {

    }

    public <T, U> boolean addEdge(Edge<Vertex<T>, Vertex<U>> e) {
        return false;
    }

    public <T> boolean removeVertex(Vertex<T> v) {
        return false;
    }

    public <T, U> boolean removeEdge(Edge<Vertex<T>, Vertex<U>> e) {
        return false;
    }

    public <T, U> boolean adjacent(Vertex<T> v, Vertex<U> w) {
        return false;
    }

    public <T> List<Vertex> neighbors(Vertex<T> v) {
        return null;
    }

    public <T> T getVertexValue(Vertex<T> v) {
        return null;
    }

    public <T> void setVertexValue(Vertex<T> v, T val) {

    }

    public <T, U> Double getEdgeValue(Edge<Vertex<T>, Vertex<U>> e) {
        return Double.NaN;
    }

    public <T, U> void setEdgeValue(Edge<Vertex<T>, Vertex<U>> e, Double val) {

    }
}
